src/Process.php

```diff
@@ -110,13 +110,24 @@ public function start(LoopInterface $loop, $interval = 0.1)
             stream_set_blocking($pipe, 0);
         }
 
-        $loop->addPeriodicTimer($interval, function (Timer $timer) {
+        $periodicMonitor = function (Timer $timer) {
             if (!$this->isRunning()) {
                 $this->close();
                 $timer->cancel();
                 $this->emit('exit', array($this->getExitCode(), $this->getTermSignal()));
             }
-        });
+        };
+
+        $timer = $loop->addPeriodicTimer($interval, $periodicMonitor);
+
+        $dataMonitor = function ($data) use ($timer, $periodicMonitor) {
+            if ('' === $data) {
+                $periodicMonitor($timer);
+            }
+        };
+
+        $this->stdout->on('data', $dataMonitor);
+        $this->stderr->on('data', $dataMonitor);
```

Comment:
The empty string check isn't an issue and is faster then calling an extra function alone on each data chunk. Like I said in my comment I really want to test this to be sure. Could be an  insignificant effect we can disregard.

Working on a pooling package build on this one, which could prove perfect to test this. I'll spin up a couple hundred child processes with it and see what the performance impact of this patch when sending a lot of data back and forth is and let you know.

Binding on the streams' `end` event could also work. Want me to test that as well while I'm at it?